src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -881,21 +881,20 @@ public virtual IHtmlContent GenerateAntiforgery(ViewContext viewContext)
                 return null;
             }
 
-            var wrappedMessage = new HtmlContentBuilder();
-            if (!string.IsNullOrEmpty(message))
+            TagBuilder messageTag;
+            if (string.IsNullOrEmpty(message))
+            {
+                messageTag = null;
+            }
+            else
             {
                 if (string.IsNullOrEmpty(headerTag))
                 {
                     headerTag = viewContext.ValidationSummaryMessageElement;
                 }
 
-                var messageTag = new TagBuilder(headerTag);
+                messageTag = new TagBuilder(headerTag);
                 messageTag.InnerHtml.SetContent(message);
-                wrappedMessage.AppendLine(messageTag);
-            }
-            else
-            {
-                wrappedMessage = null;
```

Comment:
That's right, the old code often created an `HtmlContentBuilder`, then threw it away in the next executed line. And only one entry was ever added to the builder.